tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+        var_names_for_outputs = {
+            "POS": "position",
+            "CHROM": "chrom",
+            "FILTER_PASS": "filter_pass",
+        }
+        actual_coordinates = self.test_pf7_class._add_coordinates(
+            self.test_zarr_root, True, "native", var_names_for_outputs
+        )
+        self.assertEqual(
+            list(actual_coordinates.keys()),
+            ["variant_position", "variant_chrom", "sample_id"],
+        )
+
+        values = [actual_coordinates[k] for k in actual_coordinates]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [["variants"], ["variants"], ["samples"]],
+        )
+
+    def test_add_data_vars(self):
+        var_names_for_outputs = {
+            "POS": "position",
+            "CHROM": "chrom",
+            "FILTER_PASS": "filter_pass",
+        }
+        actual_vars = self.test_pf7_class.add_data_vars(
+            self.test_zarr_root, True, "native", var_names_for_outputs
+        )
+        self.assertEqual(
+            list(actual_vars.keys()),
+            [
+                "variant_allele",
+                "variant_filter_pass",
+                "variant_is_snp",
+                "variant_numalt",
+                "variant_CDS",
+                "call_genotype",
+                "call_AD",
+            ],
+        )
+        values = [actual_vars[k] for k in actual_vars]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [
+                ["variants", "alleles"],
+                ["variants"],
+                ["variants"],
+                ["variants"],
+                ["variants"],
+                ["variants", "samples", "ploidy"],
+                ["variants", "samples", "alleles"],
+            ],
+        )
+
+    def test_add_extended_data(self):
+        test_pf7_extended = Pf7(self.test_data_path, data_config=self.test_config_path)
+        test_pf7_extended.extended_variant_fields = {"AN": [DIM_VARIANT]}
+        test_pf7_extended.extended_calldata_variables = {
+            "GQ": [DIM_VARIANT, DIM_SAMPLE]
+        }
+        actual_extended = test_pf7_extended._add_extended_data(
+            self.test_zarr_root, True, "native", {}
+        )
+        self.assertEqual(
+            list(actual_extended.keys()),
+            [
+                "call_GQ",
+                "variant_AN",
+            ],
+        )
+
+        values = [actual_extended[k] for k in actual_extended]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [
+                ["variants", "samples"],
+                ["variants"],
+            ],
+        )
```

Comment:
This tests a private method, see comment above re testing private methods.

Reply:
See comment above about testing this private method